### PR TITLE
Adds storage packing for timestamp/executed, stores the timestamp of …

### PR DESCRIPTION
**Pull request by moodysalem**
…execution, refactors the erc20 interface to include camel case (fixes https://github.com/EkuboProtocol/governance/issues/10)

### CODE REVIEW

**Comment by moodysalem** [comment]
cc @enitrat can you review this?

File: src/governance_token.cairo [hunk 1]
```diff
@@ -2,24 +2,27 @@ use core::traits::TryInto;
...
+    fn totalSupply(self: @TStorage) -> u256;
     fn balance_of(self: @TStorage, account: ContractAddress) -> u256;
+    fn balanceOf(self: @TStorage, account: ContractAddress) -> u256;
     fn allowance(self: @TStorage, owner: ContractAddress, spender: ContractAddress) -> u256;
     fn transfer(ref self: TStorage, recipient: ContractAddress, amount: u256) -> bool;
     fn transfer_from(
         ref self: TStorage, sender: ContractAddress, recipient: ContractAddress, amount: u256
     ) -> bool;
-    fn approve(ref self: TStorage, spender: ContractAddress, amount: u256) -> bool;
-    fn increase_allowance(ref self: TStorage, spender: ContractAddress, added_value: u256) -> bool;
```
Comment (enitrat) [inline src/governance_token.cairo:18]: is this related to your SNIP proposal?

File: src/governor.cairo [hunk 2]
```diff
@@ -1,26 +1,49 @@
...
+struct ProposalTimestamps {
+    // the timestamp when the proposal was created
+    creation: u64,
+    // the timestamp when the proposal was executed
+    executed: u64,
+}
+
+impl ProposalTimestampsStorePacking of StorePacking<ProposalTimestamps, u128> {
+    fn pack(value: ProposalTimestamps) -> u128 {
+        value.creation.into() + (value.executed.into() * 0x10000000000000000_u128)
```
Comment (enitrat) [inline src/governor.cairo:20]: would probably be better to have it defined as a constant
`const TWO_POW_64_MINUS_ONE: u128 = 0x10000000000000000;`
for clarity

Also generally speaking you shouldn't need to use the `_u128` suffix I think it should be inferred

**Review by enitrat — APPROVED** [review]
Looks good 🫡  Storage packing is nice

File: src/governance_token.cairo (hunk 1 above)
Comment (moodysalem) [reply src/governance_token.cairo:18]: no approve just got moved down

i removed increase allowance and decrease allowance because i think they're bad practice on starknet though, and i don't think they're even in the snip

File: src/governor.cairo (hunk 2 above)
Comment (moodysalem) [reply src/governor.cairo]: done, it's 2**64 though, no minus one
